gprofiler/main.py

```diff
@@ -851,6 +849,19 @@ def verify_preconditions(args: configargparse.Namespace) -> None:
         print("--log-usage is available only when run as a container!", file=sys.stderr)
         sys.exit(1)
 
+    if args.pids_to_profile is not None:
+        pids_to_profile = []
+        for pid in args.pids_to_profile:
+            try:
+                Process(pid)
```

Comment:
One improvement idea that I have - instead of storing the "pids to profile" in run time, you can store the `Process` objects that you create here, and then wherever you can compare `Process` object and not just pid, you can compare the object. It works better because it will handle PID reuse (and correctly refrain from profiling if PID was reused, because two Process objects are == if the PID was not reused, and != if the PID was reused, even if it's the same PID). In sites where we compare only by PID (for example, in `perf.py` where we pass the PID to profile) then you can use `Process.pid`.